src/services/posts/post.service.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  BadRequestException,
+  Inject,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
+import { DatabaseService } from '~/services/database.service'
+import { PostSubmitDto } from '~/types/post'
+import { Post } from '@prisma/client'
+
+@Injectable()
+export class PostService {
+  @Inject()
+  private readonly db!: DatabaseService
+
+  /**
+   * create a new post
+   *
+   * @param submit
+   * @param authorId
+   */
+  async create(submit: PostSubmitDto, authorId: string): Promise<Post> {
+    return await this.db.post.create({
+      data: {
+        authorId,
+        ...submit,
+      },
+    })
```

Comment:
这里的话尽量合并merge的时候注意一下顺序，如果`submit`里面也有 `authorId` 的话是会覆盖掉上面的 `authorId` 的
```suggestion
      data: {
        ...submit,
        authorId,
      },
    })
```

Reply:
😯  明白